### PR TITLE
[ENH] Tidy up loaders and writers

### LINKED ISSUE #1931: [ENH] Improvements to write_to_tsfile
### Describe the feature or idea you want to propose

just writing some data to file and will note anything here that could be better

### Describe your proposed solution

1. More informative errors
2. Data precision option: it currently creates big files
3. Doesnt work with univariate data in 2D format

### PR DESCRIPTION
**Pull request by TonyBagnall**
Fixes #1931 

Tidy up the loaders and writers. Thery dont work with df-list or multindex as yet, but if any demand for this can do it.

### CODE REVIEW

File: aeon/datasets/_data_writers.py [hunk 1]
```diff
@@ -29,6 +35,9 @@ def write_to_tsfile(
         The file is written to <path>/<problem_name>/<problem_name>.ts
     header: string, default = None
         Optional text at the top of the file that is ignored when loading.
+    suffix: str or None, default=None
+        Addon at the end of the filename before the file extension, i.e. _TRAIN or
+        _TEST
```
Comment (MatthewMiddlehurst) [inline aeon/datasets/_data_writers.py:40]: Is suffix necessary? can just be a part of `problem_name` right?

**Review by MatthewMiddlehurst — CHANGES_REQUESTED** [review]
Don't edit the changelogs

File: aeon/datasets/_data_writers.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/datasets/_data_writers.py:40]: its necessary for unequal etc

**Comment by MatthewMiddlehurst** [comment]
still an edited changelog

File: aeon/datasets/_data_writers.py (hunk 1 above)
Comment (MatthewMiddlehurst) [reply aeon/datasets/_data_writers.py:40]: I dont really see the difference between `problem_name="sample_data_eq.ts"` and `problem_name="sample_data.ts"` + `suffix="_eq" though

Comment (TonyBagnall) [reply aeon/datasets/_data_writers.py:40]: actually it was for train/test, but seeing as you object strongly, I will remove it

**Pull request merged**